go.mod

```diff
@@ -0,0 +1,13 @@
+module github.com/vshn/kube-token-refresher
+
+go 1.16
+
+require (
+	github.com/sirupsen/logrus v1.7.0
+	github.com/smartystreets/assertions v1.0.0 // indirect
+	github.com/spf13/pflag v1.0.5
+	github.com/spf13/viper v1.7.0
```

Comment:
I strongly recommend to get rid of Viper. It comes with a bloatware of unnecessary dependencies and has tons of open issues that don't seem to get fixed.
I've made good experience with https://github.com/knadh/koanf. 
Here's an example in action: https://github.com/ccremer/fronius-exporter/blob/master/cfg/config.go#L82

Reply:
While I agree that Viper is overkill for this project and pulls in a lot of dependencies, I've had good experiences with it. But I'll look into koanf, that does look interesting.

Reply:
I stand corrected. Koanf looks a lot nice :)